addons/payment/models/payment_acquirer.py

```diff
@@ -224,34 +197,34 @@ def render(self, cr, uid, id, reference, amount, currency_id, partner_id=False,
         if not values.get('partner_address'):
             values['address'] = _partner_format_address(values.get('partner_street', ''), values.get('partner_street2', ''))
         if not values.get('partner_country') and values.get('partner_country_id'):
-            values['country'] = self.pool['res.country'].browse(cr, uid, values.get('partner_country_id'), context=context)
-
+            values['country'] = self.env['res.country'].browse(values.get('partner_country_id'))
 
         # compute fees
-        fees_method_name = '%s_compute_fees' % acquirer.provider
+        fees_method_name = '%s_compute_fees' % self.provider
         if hasattr(self, fees_method_name):
-            fees = getattr(self, fees_method_name)(cr, uid, id, values['amount'], values['currency_id'], values['partner_country_id'], context=None)
-            values['fees'] = float_round(fees, 2)
+            fees = getattr(self, fees_method_name)(values['amount'], values['currency_id'], values['partner_country_id'])
+            values['fees'] = tools.float_round(fees[0], 2)
 
         # call <name>_form_generate_values to update the tx dict with acqurier specific values
-        cust_method_name = '%s_form_generate_values' % (acquirer.provider)
+        cust_method_name = '%s_form_generate_values' % (self.provider)
         if hasattr(self, cust_method_name):
             method = getattr(self, cust_method_name)
-            values = method(cr, uid, id, values, context=context)
+            inv_type = method(values)
+            values = inv_type[0] if isinstance(inv_type, list) else inv_type
 
         values.update({
-            'tx_url': context.get('tx_url', self.get_form_action_url(cr, uid, id, context=context)),
-            'submit_class': context.get('submit_class', 'btn btn-link'),
-            'submit_txt': context.get('submit_txt'),
-            'acquirer': acquirer,
-            'user': self.pool.get("res.users").browse(cr, uid, uid, context=context),
-            'context': context,
-            'type': values.get('type') or 'form',
+            'tx_url': self.env.context.get('tx_url', self.get_form_action_url()),
+            'submit_class': self.env.context.get('submit_class', 'btn btn-link'),
+            'submit_txt': self.env.context.get('submit_txt'),
+            'acquirer': self,
+            'user': self.env.user,
+            'context': self.env.context,
+            'payment_mode': values.get('payment_mode') or 'form',
         })
         values.setdefault('return_url', False)
 
         # because render accepts view ids but not qweb -> need to use the xml_id
-        return self.pool['ir.ui.view'].render(cr, uid, acquirer.view_template_id.xml_id, values, engine='ir.qweb', context=context)
+        return self.pool['ir.ui.view'].render(self._cr, self._uid, self.view_template_id.xml_id, values, context=self.env.context)  # TODO FIX: some times it creates the problem in id or xml_id
```

Comment:
@dbu-odoo  Please ```pool``` converted into the ```env```  and remove cr,uid. Like
```
self.envl['ir.ui.view'].render(self.view_template_id.xml_id, values)
```
and if really need 
```
[1] Pool
[2] cr
[3] uid
```
Then please use ```self.env``` insted of ```self._```